codis/pkg/topom/topom_group.go

```diff
@@ -640,11 +743,14 @@ func (s *Topom) SyncActionComplete(addr string, failed bool) error {
 
 	var state string
 	if !failed {
-		state = "synced"
+		state = models.ActionSynced
 	} else {
-		state = "synced_failed"
+		state = models.ActionSyncedFailed
 	}
 	g.Servers[index].Action.State = state
+	// recheck offline status through heartbeat and automatically repair
```

Comment:
Check whether the master is offline through heartbeat, if so, select a new master

Reply:
done